src/mbgl/shader/pattern.vertex.glsl

```diff
@@ -9,6 +12,6 @@ varying vec2 v_pos_b;
 
 void main() {
     gl_Position = u_matrix * vec4(a_pos, 0, 1);
-    v_pos_a = (u_patternmatrix_a * vec3(a_pos, 1)).xy;
-    v_pos_b = (u_patternmatrix_b * vec3(a_pos, 1)).xy;
+    v_pos_a = (u_patternmatrix_a * vec3(a_pos, 1)).xy + u_offset_a;
+    v_pos_b = (u_patternmatrix_b * vec3(a_pos, 1)).xy + u_offset_b;
```

Comment:
It looks to me like u_`patternmatrix_*` are transformation matrices. Have you looked into applying the offset using the transformation matrix? That'd save us from having to add `u_offset_*` uniforms.